Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -158,54 +158,242 @@ public float ScaleMinimum
         [SerializeField]
         [Tooltip("Flatten bounds in the specified axis or flatten the smallest one if 'auto' is selected")]
         private FlattenModeType flattenAxis = FlattenModeType.DoNotFlatten;
+        public FlattenModeType FlattenAxis
+        {
+            get { return flattenAxis; }
+            set
+            {
+                if (flattenAxis != value)
+                {
+                    flattenAxis = value;
+                    CreateRig();
+                }
+            }
+        }
         [SerializeField]
         [FormerlySerializedAs("wireframePadding")]
         [Tooltip("Extra padding added to the actual Target bounds")]
         private Vector3 boxPadding = Vector3.zero;
+        public Vector3 BoxPadding
+        {
+            get { return boxPadding; }
+            set
+            {
+                if (Vector3.Distance(boxPadding, value) > float.Epsilon)
+                {
+                    boxPadding = value;
+                    CreateRig();
+                }
+            }
+        }
         [SerializeField]
         [Tooltip("Material used to display the bounding box. If set to null no bounding box will be displayed")]
         private Material boxMaterial = null;
+        public Material BoxMaterial
+        {
+            get { return boxMaterial; }
+            set
+            {
+                if (boxMaterial != value)
+                {
+                    boxMaterial = value;
+                    CreateRig();
+                }
+            }
+        }
+
         [SerializeField]
         [Tooltip("Material used to display the bounding box when grabbed. If set to null no change will occur when grabbed.")]
         private Material boxGrabbedMaterial = null;
+
+        public Material BoxGrabbedMaterial
+        {
+            get { return boxGrabbedMaterial; }
+            set { boxGrabbedMaterial = value; }
+        }
+
         [SerializeField]
         [Tooltip("Show a wireframe around the bounding box when checked. Wireframe parameters below have no effect unless this is checked")]
         private bool showWireframe = true;
+
+        public bool ShowWireFrame
+        {
+            get { return showWireframe; }
+            set
+            {
+                if (showWireframe != value)
+                {
+                    showWireframe = value;
+                    CreateRig();
```

Comment:
One consequence of calling CreateRig again is that it will recalculate minimumScale/maximumScale (in CaptureInitialState). Because by default it is set relative to the current scale, if the scale of the object has changed from when it was initialized or SetScaleLimits is called, this will have an indirect consequence of changing the scale limits which is likely not intentional when showing/hiding elements.

Reply:
It looks like this may also destroy the bounding box collider override? DestroyRig destroys cachedTargetCollider, which is set from boundsOverride in SetBoundingBoxCollider.

Reply:
> It looks like this may also destroy the bounding box collider override? DestroyRig destroys cachedTargetCollider, which is set from boundsOverride in SetBoundingBoxCollider.

I read through the code and I think this one is okay. The cached target collider (which can be either created in code, or assigned to be the box collider override) only gets destroyed if it was created in code.

Reply:
> One consequence of calling CreateRig again is that it will recalculate minimumScale/maximumScale (in CaptureInitialState). Because by default it is set relative to the current scale, if the scale of the object has changed from when it was initialized or SetScaleLimits is called, this will have an indirect consequence of changing the scale limits which is likely not intentional when showing/hiding elements.

Good catch.